test/e2e/network/service.go

```diff
@@ -1736,26 +1736,60 @@ var _ = SIGDescribe("Services", func() {
 		}
 	})
 
-	ginkgo.It("should have session affinity work for service with type clusterIP", func() {
-		svc := getServeHostnameService("affinity-clusterip")
+	/*
+		Release: v1.16
+		Testname: Service, ClusterIP type, session affinity ON
+		Description: Create a service of type "ClusterIP". Service's sessionAffinity is set to "ClientIP". Service creation MUST be successful by assigning "ClusterIP" to the service.
+		Create a Replication Controller to ensure that 3 pods are running and are targeted by the service to serve hostname of the pod when requests are sent to the service.
+		Create another pod to make requests to the service. Service MUST serve the hostname from the same pod of the replica for all consecutive requests.
+		Service MUST be reachable over serviceName and the ClusterIP on servicePort.
+	*/
+	framework.ConformanceIt("should have session affinity work for service with type clusterIP", func() {
```

Comment:
These all end up calling `execAffinityTestForNonLBServiceWithOptionalTransition` which was a dense read, but AFAICT is written in a way that is cross-platform and doesn't touch any forbidden API's or features